### ISSUE TITLE
[Bug] Subgraph (non)fatal out of sync with `subgraph_errors` table

### ISSUE THREAD

**Issue by paymog** [graphprotocol/graph-node#4596]
### Bug report

I have a subgraph which has two errors in the subgraph errors table:

```
select subgraph_id, block_hash, vid, deterministic, created_at
from subgraphs.subgraph_error
where subgraph_id = 'QmSnpjbMEbFU7EJhJKSDsL6T2xNvgMcebWWcX4DpWX5u9e'

+----------------------------------------------+------------------------------------------------------------------+-----+-------------+---------------------------------+
|subgraph_id                                   |block_hash                                                        |vid  |deterministic|created_at                       |
+----------------------------------------------+------------------------------------------------------------------+-----+-------------+---------------------------------+
|QmSnpjbMEbFU7EJhJKSDsL6T2xNvgMcebWWcX4DpWX5u9e|0xCEF556098A308808ADEBE05198A41B55929E066311C18FFC601A132EBF5B0FA6|22577|true         |2023-04-26 21:26:57.088297 +00:00|
|QmSnpjbMEbFU7EJhJKSDsL6T2xNvgMcebWWcX4DpWX5u9e|0xE04C0CF8C845849843A280B3BFC805980AA81CC8D83636B9C6B7553D84FAFCEC|23260|true         |2023-05-04 14:16:15.955685 +00:00|
+----------------------------------------------+------------------------------------------------------------------+-----+-------------+---------------------------------+
```

however, when I query for this subgraph in the `subgraph_deployments` table, I see no errors

```
select deployment, fatal_error, non_fatal_errors
from subgraphs.subgraph_deployment
where deployment = 'QmSnpjbMEbFU7EJhJKSDsL6T2xNvgMcebWWcX4DpWX5u9e'


+----------------------------------------------+-----------+----------------+
|deployment                                    |fatal_error|non_fatal_errors|
+----------------------------------------------+-----------+----------------+
|QmSnpjbMEbFU7EJhJKSDsL6T2xNvgMcebWWcX4DpWX5u9e|null       |                |
+----------------------------------------------+-----------+----------------+
```

How does this edge case happen and what's the correct way to resolve this?

### IPFS hash

QmSnpjbMEbFU7EJhJKSDsL6T2xNvgMcebWWcX4DpWX5u9e

### Some information to help us out

- [ ] Tick this box if this bug is caused by a regression found in the latest release.
- [ ] Tick this box if this bug is specific to the hosted service.
- [X] I have searched the issue tracker to make sure this issue is not a duplicate.

### OS information

None

**Comment by azf20** [comment]
Thanks @paymog, what version are you running?

**Comment by paymog** [comment]
0.29.0

**Comment by incrypto32** [comment]
So in this subgraph `nonFatalErrors` feature is enabled in the manifest so fatalErrors will not get populated, the nonFatalErrors should have been though, which is fixed in #4615, so closing this issue
